src/app/components/form-fields/multiple-section-date-field/multiple-section-date-field.component.ts

```diff
@@ -168,7 +176,7 @@ export class MultipleSectionDateFieldComponent
                                 ).map((v) => v.name)
                         ],
                         override.name,
-                        override.date
+                        [override.date, courseTimeZone ?? '']
```

Comment:
Here, `courseTiemZone` is passed into the builder of `ListFieldComponent` via closure. This does not comply with `FormField`'s expected behavior of being a "pure function" of non-global context (e.g., course id & timeZone) that maps `srcValue` to `destValue`.

However, this implementation will work as well since `ListFieldComponent` is actually re-initialized each time the `srcValue` of `MultipleSectionDateFieldComponent` is set. In other words, the `srcValue` of `ListFieldComponent` in `MultipleSectionDateFieldComponent` will only be assigned once throughout its lifecycle. As a result, the change of non-global context done (e.g., changing course timeZone) by setting `srcValue` of `MultipleSectionDateFieldComponent` won't propagate the influence to `ListFieldComponent`.

If you are willing to fix this issue even though it works as expected, you can modify line `44` to include `courseTimeZone` as a part of the `srcValue` of the list item and then pass the value in at line `193`.